subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -39,6 +39,27 @@
 
 LOG_MODULE_REGISTER(bt_csip_set_member, CONFIG_BT_CSIP_SET_MEMBER_LOG_LEVEL);
 
+enum {
+	FLAG_ACTIVE,
+	FLAG_SET_MEMBER_LOCK,
+	FLAG_NUM,
+};
+
+struct csip_client {
+	bt_addr_le_t addr;
+
+/* Since there's a 1-to-1 connection between bonded devices, and devices in
+ * the array containing this struct, if the security manager overwrites
+ * the oldest keys, we also overwrite the oldest entry
+ */
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	uint32_t age;
+#endif /* CONFIG_BT_KEYS_OVERWRITE_OLDEST */
```

Comment:
Since you are refactoring this part, you can consider squeezing in the fix for https://github.com/zephyrproject-rtos/zephyr/issues/61763 as well, but up to you

Reply:
Ill take a look at this as well